internal/vpn.go

```diff
@@ -115,17 +115,30 @@ func updateConfig(app *App, client *http.Client, pubkey string) error {
 func initVpn(app *App, client *http.Client, crypto CryptoHandler) error {
 	sotaConfig := filepath.Dir(app.EncryptedConfig)
 	wgPriv := filepath.Join(sotaConfig, "wg-priv")
+	register := false
 	if _, err := os.Stat(wgPriv); os.IsNotExist(err) {
+		register = true
 		log.Println("Wireguard private key does not exist, generating.")
-		pub, err := generateKey(wgPriv)
+	} else {
+		secretFile := filepath.Join(app.SecretsDir, "wireguard-client")
+		if _, err := os.Stat(secretFile); os.IsNotExist(err) {
+			register = true
+			log.Println("Wireguard key not registered on server, will re-register")
+		}
+	}
+	if register {
+		wgPrivTmp := wgPriv + ".tmp"
+		pub, err := generateKey(wgPrivTmp)
 		if err != nil {
 			return fmt.Errorf("Unable to generate private key: %s", err)
 		}
 		log.Printf("Uploading Wireguard pub key(%s).", pub)
 		if err := updateConfig(app, client, pub); err != nil {
-			os.Remove(wgPriv)
 			return fmt.Errorf("Unable to server config with VPN public key: %s", err)
 		}
+		if err = os.Rename(wgPrivTmp, wgPriv); err != nil {
```

Comment:
If fioconfig stops before the rename, it will leave a temporary file in place and will later generate another key, is this ok from our service perspective (I guess it will just replace the older with a newer one)?

Also, maybe better to create a temporary file somewhere like /run or /tmp?

Reply:
the nice thing with `os.Rename` is that its atomic. If we use `/run`, then you cross filesystems and have to do a copy/delete operation.

Reply:
+1 for not risking the cross-filesystem nuances
A solution as is warrants atomicity and is easier to debug

Reply:
Yup, +1 for rename.